### PR TITLE
PW-4337 Update checkout API to v67 and checkout component to 4.2.1

### PR DESCRIPTION
**Pull request by peterojo**
## Summary

* update checkout client and library version
* /payment/details only takes the `details` parameter, removed `paymentData`
* remove deprecated originKey service
 
## Tested scenarios

3DS2
3DS1
iDeal
other card payments

**Fixed issue**:

### CODE REVIEW

File: src/Service/PaymentDetailsService.php [hunk 1]
```diff
@@ -68,99 +50,34 @@ class PaymentDetailsService
...
      * @return PaymentResponseHandlerResult
      * @throws PaymentFailedException
      */
-    public function doPaymentDetails(
+    public function getPaymentDetails(
         array $details,
         OrderTransactionEntity $orderTransaction
     ): PaymentResponseHandlerResult {
-        // Get paymentData for the paymentDetails request
-        $paymentResponse = $this->paymentResponseService->getWithOrderTransaction($orderTransaction);
```
Comment (peterojo) [inline src/Service/PaymentDetailsService.php:105]: paymentData is not required anymore so all the following is unnecessary

File: src/Handlers/ResultHandler.php [hunk 2]
```diff
@@ -95,71 +96,57 @@ public function __construct(
...
+        // Retrieve paymentResponse and if it exists
+        $paymentResponse = $this->paymentResponseService->getWithOrderTransaction($transaction->getOrderTransaction());
 
-        $result = $this->paymentResponseHandlerResult->createFromPaymentResponse(
-            $paymentResponse
-        );
-
-        if ('RedirectShopper' === $result->getResultCode()) {
-            // Validate 3DS1 parameters
-            // Get MD and PaRes to be validated
```
Comment (peterojo) [inline src/Handlers/ResultHandler.php:117]: `details.payload`, `details.MD` and `details.PaRes` no longer need to be sent to the `/payment/details` endpoint in v67. The only parameter we need to send here is `details.redirectResult` (`details.threeDSResult` is sent by the checkout component after 3ds2 user confirmation)
https://docs.adyen.com/online-payments/release-notes?tab=script_1666a37f_2#checkout-api-v67

File: src/Handlers/PaymentResponseHandler.php [hunk 3]
```diff
@@ -243,19 +243,24 @@ function (Context $context) use ($orderTransactionId, $customFields) {
```
Comment (peterojo) [inline src/Handlers/PaymentResponseHandler.php:243]: iDeal payments were returning with status `Pending` which was not handled here previously. Also, the exception was not logged

File: src/Handlers/PaymentResponseHandler.php [hunk 4]
```diff
@@ -243,19 +243,24 @@ function (Context $context) use ($orderTransactionId, $customFields) {
...
             case self::AUTHORISED:
+            case self::PENDING:
                 // Tag order as paid
                 $this->transactionStateHandler->paid($orderTransactionId, $context);
                 break;
             case self::REFUSED:
-                // Cancel the order
+                // Fail the order
+                $message = 'The payment was refused';
+                $this->logger->error($message, ['orderId' => $transaction->getOrder()->getId()]);
```
Comment (cyattilakiss) [inline src/Handlers/PaymentResponseHandler.php:253]: I think it's more like an info than an error

File: src/Handlers/PaymentResponseHandler.php [hunk 5]
```diff
@@ -243,19 +243,24 @@ function (Context $context) use ($orderTransactionId, $customFields) {
...
                 throw new PaymentFailedException(
-                    'The payment was refused'
+                    $message
                 );
                 break;
             case self::CANCELLED:
                 // Cancel the order
+                $message = 'The payment was cancelled';
+                $this->logger->error($message, ['orderId' => $transaction->getOrder()->getId()]);
```
Comment (cyattilakiss) [inline src/Handlers/PaymentResponseHandler.php:261]: I think it's more like an info than an error

File: src/Handlers/ResultHandler.php [hunk 6]
```diff
@@ -95,61 +96,43 @@ public function __construct(
...
-            if (!empty($redirectResult)) {
-                $details[self::REDIRECT_RESULT] = $redirectResult;
-            }
-
-            if (!empty($payload)) {
-                $details[self::PAYLOAD] = $payload;
-            }
+        if ('RedirectShopper' === $result->getResultCode()) {
+            $redirectResult = $request->query->get(
+                self::REDIRECT_RESULT,
```
Comment (cyattilakiss) [inline src/Handlers/ResultHandler.php:118]: is it always only self::REDIRECT_RESULT? (at least for the current payment methods that we do support)

File: src/Handlers/ResultHandler.php [hunk 7]
```diff
@@ -95,61 +96,43 @@ public function __construct(
...
                 $error = 'Payment details are missing.';
                 $this->logger->error(
                     $error,
                     ['orderId' => $transaction->getOrder()->getId()]
                 );
                 throw new PaymentFailedException($error);
             }
+            $details = [];
+
+            $details[self::REDIRECT_RESULT] = $redirectResult;
```
Comment (cyattilakiss) [inline src/Handlers/ResultHandler.php:132]:
```suggestion
            $details = [
                self::REDIRECT_RESULT => $redirectResult
            ];
```

**Review by cyattilakiss — COMMENTED** [review]
Did you newly generate the 4.2.1 bundle?

File: src/Handlers/ResultHandler.php (hunk 6 above)
Comment (peterojo) [reply src/Handlers/ResultHandler.php:118]: Yes, `payload` was replaced in [v66](https://docs.adyen.com/online-payments/release-notes?tab=npm-recommended-_1666a37f_1#checkout-api-v66) for 
> redirects, except 3D Secure redirects

and then in [v67](https://docs.adyen.com/online-payments/release-notes?tab=script_1666a37f_2#checkout-api-v67):
>For redirect 3D Secure authentication, you now need to submit the redirectResult appended to the returnUrl when the shopper is redirected back to you. This replaces the `MD` and the `PaRes` parameters used in previous versions, and aligns the flow for 3D Secure redirects with all other redirects.

**Comment by peterojo** [comment]
> Did you newly generate the 4.2.1 bundle?

@cyattilakiss yes I did

**Review by cyattilakiss — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Pull request merged**